### DISCUSSION TITLE
Cannot cast Optional property to int

### DISCUSSION THREAD

**Discussion by ludo237** [spatie/laravel-data#275] [Q&A]
Hello everyone.
I've a simple DTO like the following example

```php
class UpdateUserData extends Data
{
    public function __construct(
        #[Sometimes]
        public Optional|int $age,
     ){}
}
```

This cast doesn't like the fact that my Eloquent model has an internal cast for age as follow

```php
class User extends Model
{
    protected $casts = [
        "age" => "integer",
    ];
}
```

If I try to use that dto this is the error that comes out 
```bash
Object of class Spatie\LaravelData\Optional could not be converted to int
at vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php:753
```

I honestly have no clue what to do next, do I need to drop Eloquent cast? I would like to keep the Eloquent internal cast so my next idea could be to create a custom Eloquent cast for integers but that sounds weird. 

**UPDATE** This seems to be a problem with Eloquent `$casts` in general not only with integers

**Comment by alexrififi** [comment]
> If I try to use...

How?

**Reply by ludo237** [reply]
Right now I'm using laravel-data as DTO package so usually I have them inside my controller like so `UpdateUserData::fromRequest($request)`. DTO has also the `rules()` method that I've omitted here for simplicity

**Reply by alexrififi** [reply]
What's in the `fromRequest` method?

**Comment by rubenvanassche** [comment]
Looks like somehow the property was not filled in, since the optional value was used. How did you create the data object?
